framer/InputLayer.coffee

```diff
@@ -0,0 +1,151 @@
+Utils = require "./Utils"
+
+{Layer} = require "./Layer"
+{TextLayer} = require "./TextLayer"
+{Events} = require "./Events"
+
+Events.EnterKey  = "EnterKey"
+Events.BackSpaceKey  = "BackSpaceKey"
+Events.InputValueChange = "InputValueChange"
+Events.InputFocus = "InputFocus"
+Events.InputBlur = "InputBlur"
+
+class exports.InputLayer extends TextLayer
+
+	constructor: (options={}) ->
+
+		_.defaults options,
+			backgroundColor: "#FFF"
+			width: 500
+			height: 100
+
+		super options
+
+		# Layer containing input element
+		@input = new Layer
+			backgroundColor: "transparent"
+			x: 30
+			width: @width - 60
+			height: @height
+			parent: @
+			name: "input"
+
+		if not @multiLine
+			@_inputElement = document.createElement("input")
+		else
+			@_inputElement = document.createElement("textarea")
+			@input.y = 25
+
+		# The id serves to differentiate multiple input elements from one another.
+		# To allow styling the placeholder colors of seperate elements.
+		@_id = Utils.round(Utils.randomNumber(0, 100))
+		@_inputElement.className = "input" + @_id
+
+		@on "change:width", =>
+			@_inputElement.style.width = "#{@input.width}px}"
+
+		# Append element
+		@input._element.appendChild(@_inputElement)
+
+		# Match TextLayer defaults and type properties
+		@_setTextProperties = =>
+			@_inputElement.style.fontFamily = @fontFamily
+			@_inputElement.style.fontSize = @fontSize
+			@_inputElement.style.lineHeight = @lineHeight
+			@_inputElement.style.fontWeight = @fontWeight
+			@_inputElement.style.color = @color
+			@_inputElement.style.outline = "none"
+			@_inputElement.style.backgroundColor = "transparent"
+			@_inputElement.style.width = "#{@width - 64}px"
+			@_inputElement.style.height = "#{@height}px"
+			@_inputElement.style.cursor = "auto"
+			@_inputElement.style.webkitAppearance = "none"
+			@_inputElement.style.resize = "none"
+
+		@_setTextProperties()
+
+		# All inherited properties
+		@_textProperties =
+			{@fontFamily, @fontSize, @lineHeight, @fontWeight, @color, @backgroundColor, @width, @height}
+
+		for property, value of @_textProperties
+
+			@on "change:#{property}", (value) =>
+				@_setTextProperties()
+				@_setPlaceholder()
+
+		# If text has been defined, use that, otherwise default to placeholder
+		@_defaultText = @text
+		@_setPlaceholder()
+
+		# Override text property setting the html
+		@html = ""
+
+		# Check if in focus
+		@_isFocused = false
+
+		# Default focus interaction
+		@_inputElement.onfocus = (e) =>
+
+			@_inputElement.style.color = "#000"
```

Comment:
It is! Wasn't using the `focusColor` property here though. Fixed and rewritten in https://github.com/koenbok/Framer/commit/36c6eb8d4e33f40ada5384fc5d4333d211f5ddcb